From d5f03f713d3ec1ab4530066d51190cf59dcba952 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 17 Feb 2022 19:52:05 +0200
Subject: [PATCH] [SW-0] gaudi3: export get rom hash function

hash stored in ROM is copied into PSOC_SECURITY block key register.
Since PSOC_SECURITY block registers are not privileged, FW cannot
use this hash for any image verification as the values stored in
those hash registers are for debug purpose only. In order for
other FW components to retrieve hash stored in ROM, this patch
exports rom_get_hash function.

Change-Id: I29ea22ad7646bef852d5c3c2d4ffaec943601f83
---

diff --git a/arch/gaudi3/arc/arcboot_rom.txt b/arch/gaudi3/arc/arcboot_rom.txt
index 7f81c2c..8fbe384 100644
--- a/arch/gaudi3/arc/arcboot_rom.txt
+++ b/arch/gaudi3/arc/arcboot_rom.txt
@@ -5,3 +5,4 @@
 dma_copy_image
 spi_flash_readl
 dma_copy_image_v2
+security_rom_get_hash
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 8382413..5318910 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -143,7 +143,7 @@
 	return -1;
 }
 
-static void security_rom_get_hash(uint32_t *key_hash)
+void security_rom_get_hash(uint32_t *key_hash)
 {
 	uint32_t *ptr = (uint32_t *)rom_key_hashes;
 	int size = NUM_KEYS_IN_ROM * SECURITY_DWORD_PER_KEY;
